sources/editor/Stride.GameStudio.Avalonia/ViewModels/MainViewModel.cs

```diff
@@ -138,6 +142,140 @@ private void OnExit()
         DialogService.Exit();
     }
 
+    private bool BuildProject(string projectPath, string framework, string workingDirectory)
+    {
+        var process = new Process
+        {
+            StartInfo = new ProcessStartInfo
+            {
+                FileName = "dotnet",
+                Arguments = $"build \"{projectPath}\" --framework {framework}",
+                WorkingDirectory = workingDirectory,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true,
+            }
+        };
+
+        process.OutputDataReceived += (s, e) => { if (e.Data != null) Console.WriteLine("[build] " + e.Data); };
+        process.ErrorDataReceived += (s, e) => { if (e.Data != null) Console.Error.WriteLine("[build-err] " + e.Data); };
+
+        try
+        {
+            process.Start();
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+            process.WaitForExit();
```

Comment:
I'd prefer to use the async version.

Reply:
See: https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexitasync